### ISSUE TITLE
Improve export: filename and extension

### ISSUE THREAD

**Issue by fam4r** [koelleChristian/trickytripper#36]
Hi,

export function seems not to work in the same way for different applications.
I tested: Telegram, WhatsApp, MEGA, Outlook, Tutanota, Protonmail, and Email (LineageOS email app)

- Telegram: exported and succesfully sent file as `File`, without extension
- WhatsApp: exported and succesfully sent file as `Untitled`, without extension
- MEGA: opening folder chooser, but not uploading file at all
- Outlook: creates email draft and attaches a file but with `attachment` name and without extension
- Tutanota: not exporting file neither creating a new email draft at all: it opens email list
- Protonmail: creates email draft but not attaches nothing
- Email: creates email draft and attaches a file with meaningful name and extension (expected behavior?)

Info: F-Droid build. Version: 1.5.9
Android 8.1

Thank you, the app is great and **very** useful

**Comment by koelleChristian** [comment]
Well ...
A long time ago, I implemented the report export for specific applications only. The world was shocked, as I did not support everyone's favorite apps, although it totally suited my needs.

Now, I  have implemented an generic export which works this way:
* TT tells the OS: "I have a file to offer for all applications which claim to be able to handle it."
* The OS then shows a popup with all applications which claim to be able to handle an attachment.
* On my phone I get more applications non-working applications offered as a export target, than working ones, however, I know which one to pick (K9, Dropbox)
* There might be a bug in my implementation, and there might be tweaks and proprietary interfaces to get the applications mentioned above working, However. I will not investigate the ones specifically mentioned by you, as for example WhatsApp would be my last choice of application.

The TT source code is free, so feel free to investigate them on your own and submit a pull request.
